src/firefly/js/ui/ImageSelectDropdown.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+
+
+// ----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+// THIS PANEL IS TEMPORARY, ONLY TO TEST CATALOGS UNTIL WE FINISH THE REAL PANEL
+// This panel will do search on 3 of the most common IRSA catalogs
+// ----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+// ----------------------------------------------------------------------------
+
+import React, { Component, PropTypes} from 'react';
+import {flux} from '../Firefly.js';
+
+import FormPanel from './FormPanel.jsx';
+import {dispatchHideDropDownUi} from '../core/LayoutCntlr.js';
+import FieldGroupUtils from '../fieldGroup/FieldGroupUtils';
+import {ImageSelection, getPlotInfo, panelKey, isCreatePlot} from '../visualize/ui/ImageSelectPanel.jsx';
+import {resultSuccess, resultFail} from '../visualize/ui/ImageSelectPanelResult.js';
+
+const dropdownName = 'ImageSelectDropDownCmd';
+/**
+ *
+ * @param props
+ * @return {XML}
+ * @constructor
+ */
+export class ImageSelectDropdown extends Component {
+    constructor(props) {
+        super(props);
+
+        this.plotInfo = getPlotInfo();
+        this.state = {
+            addPlot: false
+        };
+    }
+
+
+    componentWillUnmount() {
+        this.iAmMounted= false;
+        if (this.removeListener) this.removeListener();
+    }
+
+    componentDidMount() {
+        this.iAmMounted = true;
+        this.removeListener = flux.addListener(() => this.stateUpdate());
+
+    }
+
+    stateUpdate() {
+        var fields = FieldGroupUtils.getGroupFields(panelKey);
+
+        if (fields && this.iAmMounted ) {
+             this.setState({addPlot: isCreatePlot(this.plotInfo.plotMode, fields)});
```

Comment:
'fields' is used to calculate the value of 'addPlot' for the state. Not very sure the meaning of  'need it in the state'? If I have any misunderstanding, please let me know again. Thanks.